Add option for default reply privacy set to unlisted by default

fadookie opened the pull request:
This PR fixes https://github.com/tuskyapp/Tusky/issues/2798 and is mostly based on and supersedes https://github.com/tuskyapp/Tusky/pull/2826 but I have fixed all merge conflicts and unit tests.

I tested the changes locally and the setting takes effect immediately for replies, and persists across killing the app.

connyduck commented:
It now looks as if this setting was synced with the server, which it can't be because Mastodon doesn't have this feature.

<img src="https://github.com/tuskyapp/Tusky/assets/10157047/9b7c267b-159d-4355-acbb-ecc430dc8323" width="320"/>

Some ideas how we could solve this:

- Move the (synced with server) out of the section title and into each setting title where it is true.
- Move the new setting into a new section "Publishing (local only)" (or something like that.

thoughs? 🤔

fadookie commented:
@connyduck Either is fine with me, conceptually I think it makes a bit more sense to be right where I've put it so it's right next to default post privacy. So maybe the former option you proposed is better?*

connyduck commented:
Ok, can you change the strings please?

fadookie commented:
@connyduck Okay, I've made the updates you've requested. I had to make all these settings labels multi-line so the text would fit - look good?
![Screenshot_1717882942](https://github.com/tuskyapp/Tusky/assets/110764/0f58c6b4-29ed-4c4d-99a2-a4de2faa1952)

I also cleaned up the history a little, removed some unnecessary merge commits, GPG signed and tagged the correct issue in the commit messages.

connyduck commented on app/src/main/java/com/keylesspalace/tusky/di/StorageModule.kt:
```diff
@@ -65,8 +66,11 @@ object StorageModule {
                 AppDatabase.MIGRATION_44_45, AppDatabase.MIGRATION_45_46, AppDatabase.MIGRATION_46_47,
                 AppDatabase.MIGRATION_47_48, AppDatabase.MIGRATION_52_53, AppDatabase.MIGRATION_54_56,
-                AppDatabase.MIGRATION_58_60
+                AppDatabase.MIGRATION_58_60, AppDatabase.MIGRATION_58_60
```
```suggestion
                AppDatabase.MIGRATION_58_60, AppDatabase.MIGRATION_60_62
```

fadookie replied:
Oops, I had to redo this merge when I was cleaning up the history and made a mistake. Good catch.

connyduck commented on app/src/main/java/com/keylesspalace/tusky/di/StorageModule.kt:
```diff
@@ -65,8 +66,11 @@ object StorageModule {
     }
+
+    @Provides
+    fun providesAccountDao(db: AppDatabase): AccountDao = db.accountDao()
```
But `AccountDao` isn't injected anywhere? Either remove this again or actually inject it somewhere.

fadookie replied:
Good catch. This is an artifact of the merge resolution I did because `AccountDao` is now being retrieved from the DB object instead of injected. Seems simplest to remove the provider for now since we don't really need to mock it in the unit tests, I am mocking the account manager instead.
